openmc/deplete/microxs.py

```diff
@@ -0,0 +1,178 @@
+                              '(n,4n)',
+                              'fission'],
+                   energy_bounds=(0, 20e6)):
+        """Generate a one-group cross-section dataframe using
+        OpenMC. Note that the ``openmc`` executable must be compiled.
+
+        Parameters
+        ----------
+        model : openmc.Model
+            OpenMC model object. Must contain geometry, materials, and settings.
+        reaction_domain : openmc.Material or openmc.Cell or openmc.Universe or openmc.RegularMesh
+            Domain in which to tally reaction rates.
+        reactions : list of str, optional
+            Reaction names to tally
+        energy_bound : 2-tuple of float, optional
+            Bounds for the energy group.
+
+        Returns
+        -------
+        MicroXS, cross section data in [b]
+
+        """
+        groups = EnergyGroups(energy_bounds)
+
+        # Set up the reaction tallies
+        original_tallies = model.tallies
+        tallies = Tallies()
+        xs = {}
+        for rxn in reactions:
+            if rxn == 'fission':
+                xs[rxn] = FissionXS(domain=reaction_domain, groups=groups, by_nuclide=True)
+            else:
+                xs[rxn] = ArbitraryXS(rxn, domain=reaction_domain, groups=groups, by_nuclide=True)
+            tallies += xs[rxn].tallies.values()
+
+        model.tallies = tallies
+
+        # create temporary run
+        original_dir = Path.cwd()
+        temp_dir = tempfile.TemporaryDirectory()
+        chdir(tempfile.gettempdir())
+        statepoint_path = model.run()
+        chdir(original_dir)
+
+        with StatePoint(statepoint_path) as sp:
+            for rxn in xs:
+                xs[rxn].load_from_statepoint(sp)
+
+        # Build the DataFrame
+        micro_xs = cls()
+        for rxn in xs:
+            df = xs[rxn].get_pandas_dataframe(xs_type='micro')
+            df.index = df['nuclide']
+            df.drop(['nuclide', xs[rxn].domain_type, 'group in', 'std. dev.'], axis=1, inplace=True)
+            df.rename({'mean':rxn}, axis=1, inplace=True)
+            micro_xs = concat([micro_xs, df], axis=1)
+
+        micro_xs._units = 'b'
+
+        # Revert to the original tallies
+        model.tallies = original_tallies
+
+        return micro_xs
+
+    @classmethod
+    def from_array(cls, nuclides, reactions, data, units='b'):
+        """
+        Creates a ``MicroXS`` object from arrays.
+
+        Parameters
+        ----------
+        nuclides : list of str
+            List of nuclide symbols for that have data for at least one
+            reaction.
+        reactions : list of str
+            List of reactions. All reactions must match those in
+            :data:`openmc.deplete.chain.REACTIONS`
+        data : ndarray of floats
+            Array containing one-group microscopic cross section values for
+            each nuclide and reaction
+        units : {'b', 'cm^2'}
+            Units of the cross section values in ``data``
```

Comment:
I added it back in because the `from_arrays` and `from_csv` methods have no way to actually check the units of the data. But I can remove it if you think it isn't useful.